src/osd/osd_types.cc

```diff
@@ -276,32 +276,48 @@ void request_redirect_t::generate_test_instances(list<request_redirect_t*>& o)
 
 void objectstore_perf_stat_t::dump(Formatter *f) const
 {
-  f->dump_unsigned("commit_latency_ms", os_commit_latency);
-  f->dump_unsigned("apply_latency_ms", os_apply_latency);
+  // *_ms values just for compatibility.
+  f->dump_float("commit_latency_ms", os_commit_latency_ns / 1000000.0);
+  f->dump_float("apply_latency_ms", os_apply_latency_ns / 1000000.0);
+  f->dump_unsigned("commit_latency_ns", os_commit_latency_ns);
+  f->dump_unsigned("apply_latency_ns", os_apply_latency_ns);
 }
 
 void objectstore_perf_stat_t::encode(bufferlist &bl) const
 {
-  ENCODE_START(1, 1, bl);
-  encode(os_commit_latency, bl);
-  encode(os_apply_latency, bl);
+  uint32_t commit_latency_ms = os_commit_latency_ns / 1000000;
+  uint32_t apply_latency_ms = os_apply_latency_ns / 1000000;
+  ENCODE_START(2, 1, bl);
+  encode(commit_latency_ms, bl); // for compatibility with older monitor.
+  encode(apply_latency_ms, bl); // for compatibility with older monitor.
+  encode(os_commit_latency_ns, bl);
+  encode(os_apply_latency_ns, bl);
   ENCODE_FINISH(bl);
 }
 
 void objectstore_perf_stat_t::decode(bufferlist::iterator &bl)
 {
-  DECODE_START(1, bl);
-  decode(os_commit_latency, bl);
-  decode(os_apply_latency, bl);
+  DECODE_START(2, bl);
+  uint32_t commit_latency_ms;
+  uint32_t apply_latency_ms;
+  decode(commit_latency_ms, bl);
+  decode(apply_latency_ms, bl);
+  if (struct_v >= 2) {
+      decode(os_commit_latency_ns, bl);
```

Comment:
wrong indent.

Reply:
Is there any auto-indentation script ?

Reply:
@socketpair not yet, AFAICT. some developers (it not all) are using VIM or Emacs for editing source files. and these editors read the file variables at the first lines. for the coding style we are following, see https://github.com/ceph/ceph/blob/master/CodingStyle

Reply:
Fixed. FYI:

`clang-format-3.9 -i -style="{BasedOnStyle: LLVM, IndentWidth: 2, BreakBeforeBraces: Linux, AllowShortIfStatementsOnASingleLine: false, IndentCaseLabels: false}"` makes huge amount of changes. I think, someone should reformat sources using this (or another) tool and also add github check that patched sources do not change after automatic reformatting.